library/ssl_cli.c

```diff
@@ -1088,12 +1088,21 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
         mbedtls_ssl_send_flight_completed( ssl );
 #endif
 
-    if( ( ret = mbedtls_ssl_write_record( ssl ) ) != 0 )
+    if( ( ret = mbedtls_ssl_write_handshake_msg( ssl ) ) != 0 )
     {
-        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_write_record", ret );
+        MBEDTLS_SSL_DEBUG_RET( 1, "mbedtls_ssl_write_handshake_msg", ret );
         return( ret );
     }
 
+#if defined(MBEDTLS_SSL_PROTO_DTLS)
```

Comment:
This code is used once here, and twice in the server code - it could be extracted into a shared function.

Reply:
I agree, and it should be subsumed with `mbedtls_ssl_send_flight_completed()`, because they always go in tandem - however, this is coming from #1939, and it is not unlikely that this code might go away with MPS anyway, so I'd be inclined to leave things as they are for now.